packages/openapi-fetch/test/index.test.ts

```diff
@@ -1,33 +1,137 @@
-// @ts-expect-error
-import createFetchMock from "vitest-fetch-mock";
+import {
+  http,
+  HttpResponse,
+  type JsonBodyType,
+  type StrictResponse,
+  type StrictRequest,
+  type DefaultBodyType,
+  type HttpResponseResolver,
+  type PathParams,
+  type AsyncResponseResolverReturnType,
+} from "msw";
+import { setupServer } from "msw/node";
 import createClient, {
   type Middleware,
   type MiddlewareRequest,
   type QuerySerializerOptions,
 } from "../src/index.js";
 import type { paths } from "./fixtures/api.js";
 
-const fetchMocker = createFetchMock(vi);
+const server = setupServer();
+
+/**
+ * Default baseUrl for tests
+ */
+const baseUrl = "https://api.example.com";
+
+/**
+ * Test path helper, returns a an absolute URL based on
+ * the given path and base
+ */
+function testPath(path: string, base: string = baseUrl) {
+  // If we have absolute path
+  if (URL.canParse(path)) {
+    return new URL(path).toString();
+  }
+
+  // Otherwise we want to support relative paths
+  // where base may also contain some part of the path
+  // e.g.
+  // base = https://api.foo.bar/v1/
+  // path = /self
+  // should result in https://api.foo.bar/v1/self
+
+  // Construct base URL
+  const baseUrlInstance = new URL(base);
+
+  // prepend base url url pathname to path and ensure only one slash between the URL parts
+  const newPath = `${baseUrlInstance.pathname}/${path}`.replace(/\/+/g, "/");
+
+  return new URL(newPath, baseUrlInstance).toString();
+}
 
 beforeAll(() => {
-  fetchMocker.enableMocks();
-});
-afterEach(() => {
-  fetchMocker.resetMocks();
+  server.listen({
+    onUnhandledRequest: (request) => {
+      throw new Error(
+        `No request handler found for ${request.method} ${request.url}`,
+      );
+    },
+  });
 });
 
-interface MockResponse {
+afterEach(() => server.resetHandlers());
+
+afterAll(() => server.close());
+
+type MswHttpMethod = keyof typeof http;
+
+interface TestRequestHandlerOptions<
+  // Recreate the generic signature of the HTTP resolver
+  // so the arguments passed to http handlers propagate here.
+  Params extends PathParams<keyof Params> = PathParams,
+  RequestBodyType extends DefaultBodyType = DefaultBodyType,
+  ResponseBodyType extends DefaultBodyType = undefined,
+> {
+  baseUrl?: string;
+  method: MswHttpMethod;
+  /**
+   * Relative or absolute path to match.
+   * When relative, baseUrl will be used as base.
+   */
+  path: string;
+  body?: JsonBodyType;
   headers?: Record<string, string>;
-  status: number;
-  body: any;
+  status?: number;
+  handler?: HttpResponseResolver<Params, RequestBodyType, ResponseBodyType>;
 }
 
-function mockFetch(res: MockResponse) {
-  fetchMocker.mockResponse(() => res);
-}
+function useTestRequestHandler<
```

Comment:
Not sure if this is the best name either, naming things is hard :).

Possible alternatives: `useMockRequestHandler`, `mockRequestHandler`, `mockHandler`

Reply:
This (and all the other names) are fine! Better than the current versions anyway